In [0]:
import os
import pandas as pd
from google.colab import drive
drive.mount('/content/Drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&scope=email%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdocs.test%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive.photos.readonly%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fpeopleapi.readonly&response_type=code

Enter your authorization code:
··········
Mounted at /content/Drive


In [0]:
cd "/content/Drive/My Drive/Spring 2019/ML/sample for project"

/content/Drive/My Drive/Spring 2019/ML/sample for project


In [0]:
small_df = pd.read_csv("image_info.csv")

In [0]:
small_df.head()

PostImage  person  bicycle  car  \
0  clickbait17-train-170331/media/608744358184108...       3        0    0   
1  clickbait17-train-170331/media/608749345731674...       2        0    0   
2  clickbait17-train-170331/media/608743563082493...       4        0    0   
3  clickbait17-train-170331/media/608744358184108...       3        0    0   
4  clickbait17-train-170331/media/608749345731674...       2        0    0   

   motorcycle  airplane  bus  train  truck  boat  ...  sink  refrigerator  \
0           0         0    0      0      0     0  ...     0             0   
1           0         0    0      0      0     0  ...     0             0   
2           0         0    0      0      0     0  ...     0             0   
3           0         0    0      0      0     0  ...     0             0   
4           0         0    0      0      0     0  ...     0             0   

   Unnamed: 83  book  clock  vase  scissors  teddy bear  hair drier  \
0            0     0      0     0         0           0           0   
1            0     0      0     0         0           0           0   
2            0     0      0     0         0           0           0   
3            0     0      0     0         0           0           0   
4            0     0      0     0         0           0           0   

   toothbrush  
0           0  
1           0  
2           0  
3           0  
4           0  

[5 rows x 91 columns]

In [0]:
type(small_df['PostImage'][0])

str

In [0]:
df1

PostImage  person  bicycle  car  \
0     clickbait17-train-170331/media/608744358184108...       3        0    0   
1     clickbait17-train-170331/media/608749345731674...       2        0    0   
2     clickbait17-train-170331/media/608743563082493...       4        0    0   
6     clickbait17-train-170331/media/608744463796609...       0        0    0   
7     clickbait17-train-170331/media/608739407592423...       3        0    0   
8     clickbait17-train-170331/media/608742768882647...       3        0    0   
9     clickbait17-train-170331/media/608747894808702...       3        0    0   
10    clickbait17-train-170331/media/608744232942067...       0        0    0   
11    clickbait17-train-170331/media/608739044500107...       1        0    0   
12    clickbait17-train-170331/media/608743812333236...       0        0    0   
13    clickbait17-train-170331/media/608739280559722...       0        0    1   
14    clickbait17-train-170331/media/608743659668955...       2        0    0   
15    clickbait17-train-170331/media/608750006674165...       0        0    0   
16    clickbait17-train-170331/media/608762951227592...      12        0    0   
17    clickbait17-train-170331/media/608761340077838...       3        0    0   
18    clickbait17-train-170331/media/608750189298327...       5        0    0   
19    clickbait17-train-170331/media/608759561541935...       1        0    0   
20    clickbait17-train-170331/media/608756156476076...       0        0    0   
21    clickbait17-train-170331/media/608755591729668...       0        0    0   
22    clickbait17-train-170331/media/608751024325689...       0        0    0   
23    clickbait17-train-170331/media/608759566898036...       3        0    0   
24    clickbait17-train-170331/media/608762127260749...       1        0    0   
25    clickbait17-train-170331/media/608751950566404...       3        0    0   
26    clickbait17-train-170331/media/608759317190221...       4        0    0   
27    clickbait17-train-170331/media/608755787540750...       5        0    0   
28    clickbait17-train-170331/media/608773560396423...       4        0    0   
29    clickbait17-train-170331/media/608775432268337...       1        0    0   
30    clickbait17-train-170331/media/608765779908022...       4        0    1   
31    clickbait17-train-170331/media/608788996445986...       1        0    0   
32    clickbait17-train-170331/media/608778182188257...       0        0    0   
...                                                 ...     ...      ...  ...   
1596  clickbait17-train-170331/media/608704949422276...       1        0    0   
1597  clickbait17-train-170331/media/608702957048225...       6        0    0   
1598  clickbait17-train-170331/media/608710272379830...       2        0    0   
1599  clickbait17-train-170331/media/608707065452867...       3        0    0   
1600  clickbait17-train-170331/media/608710504148631...       4        0    0   
1601  clickbait17-train-170331/media/608708086220455...       1        0    0   
1602  clickbait17-train-170331/media/608707023258161...       1        0    8   
1603  clickbait17-train-170331/media/608706198217568...       0        0    0   
1604  clickbait17-train-170331/media/608724236035710...       0        0    0   
1605  clickbait17-train-170331/media/608714030937604...       2        0    0   
1606  clickbait17-train-170331/media/608724691491123...       5        0    0   
1607  clickbait17-train-170331/media/608724295863427...       0        0    0   
1608  clickbait17-train-170331/media/608717808344551...      12        0    0   
1609  clickbait17-train-170331/media/608717827470573...       1        0    0   
1610  clickbait17-train-170331/media/608722692779425...       1        0    0   
1611  clickbait17-train-170331/media/608717216758964...       1        0    0   
1612  clickbait17-train-170331/media/608711317952724...       0        0    0   
1613  clickbait17-train-170331/media/608720863953842...       7        0    1

In [0]:
small_df['PostImage'][0]

'clickbait17-train-170331/media/608744358184108032.jpg'

In [0]:
s = df1.loc[0]['PostImage']
s

'/media/608744358184108032.jpg'

In [0]:
# for col,row in df1.iterrows():
#   tmp = row['PostImage']
#   tmp = tmp.replace('clickbait17-train-170331','')
#   row['PostImage'] = tmp
#   print (row['PostImage'])
for i in range(len(small_df)):
  tmp = small_df['PostImage'][i]
  tmp = tmp.replace('clickbait17-train-170331','')
  small_df.set_value(i,'PostImage', tmp)


/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:4: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
  after removing the cwd from sys.path.


In [0]:
small_df

PostImage  person  bicycle  car  motorcycle  \
0     /media/608744358184108032.jpg       3        0    0           0   
1     /media/608749345731674112.png       2        0    0           0   
2     /media/608743563082493953.jpg       4        0    0           0   
3     /media/608744358184108032.jpg       3        0    0           0   
4     /media/608749345731674112.png       2        0    0           0   
5     /media/608743563082493953.jpg       4        0    0           0   
6     /media/608744463796609024.png       0        0    0           0   
7     /media/608739407592423424.jpg       3        0    0           0   
8     /media/608742768882647040.jpg       3        0    0           0   
9     /media/608747894808702978.png       3        0    0           0   
10    /media/608744232942067712.jpg       0        0    0           0   
11    /media/608739044500107264.png       1        0    0           0   
12    /media/608743812333236225.jpg       0        0    0           0   
13    /media/608739280559722498.jpg       0        0    1           0   
14    /media/608743659668955137.png       2        0    0           0   
15    /media/608750006674165760.jpg       0        0    0           0   
16    /media/608762951227592704.jpg      12        0    0           0   
17    /media/608761340077838336.png       3        0    0           0   
18    /media/608750189298327552.jpg       5        0    0           0   
19    /media/608759561541935104.png       1        0    0           0   
20    /media/608756156476076032.jpg       0        0    0           0   
21    /media/608755591729668097.jpg       0        0    0           0   
22    /media/608751024325689345.png       0        0    0           0   
23    /media/608759566898036736.png       3        0    0           0   
24    /media/608762127260749825.jpg       1        0    0           0   
25    /media/608751950566404096.jpg       3        0    0           0   
26    /media/608759317190221826.jpg       4        0    0           0   
27    /media/608755787540750337.png       5        0    0           0   
28    /media/608773560396423169.jpg       4        0    0           0   
29    /media/608775432268337152.jpg       1        0    0           1   
...                             ...     ...      ...  ...         ...   
1596  /media/608704949422276608.png       1        0    0           0   
1597  /media/608702957048225793.jpg       6        0    0           0   
1598  /media/608710272379830273.png       2        0    0           0   
1599  /media/608707065452867584.png       3        0    0           0   
1600  /media/608710504148631552.jpg       4        0    0           0   
1601  /media/608708086220455936.png       1        0    0           2   
1602  /media/608707023258161152.jpg       1        0    8           0   
1603  /media/608706198217568256.jpg       0        0    0           0   
1604  /media/608724236035710979.jpg       0        0    0           0   
1605  /media/608714030937604096.jpg       2        0    0           0   
1606  /media/608724691491123200.png       5        0    0           0   
1607  /media/608724295863427072.jpg       0        0    0           0   
1608  /media/608717808344551424.jpg      12        0    0           0   
1609  /media/608717827470573568.jpg       1        0    0           0   
1610  /media/608722692779425792.jpg       1        0    0           0   
1611  /media/608717216758964224.jpg       1        0    0           0   
1612  /media/608711317952724992.jpg       0        0    0           0   
1613  /media/608720863953842179.png       7        0    1           0   
1614  /media/608723346965061632.jpg       2        0    0           0   
1615  /media/608736668347039744.jpg       2        0    0           0   
1616  /media/608730313926180865.jpg       1        0    0           0   
1617  /media/608733156674555904.jpg       1        0    0           0   
1618  /media/608727096098078720.jpg       0        0    0           0   
1619  /medi

In [0]:
small_df.duplicated().sum()

3

In [0]:
len(small_df)

1626

In [0]:
df1 = small_df.drop_duplicates()

In [0]:
df1.shape

(1623, 91)

In [0]:
big_df = pd.read_csv('big_info.csv')

In [0]:
big_df.duplicated().sum()

5

In [0]:
len(big_df)

11046

Number of Image 11,067 = 11046(num of big_df) + 23 (dump) - 5 (duplicated)

In [0]:
for i in range(len(big_df)):
  tmp = big_df['PostImage'][i]
  tmp = tmp.replace('clickbait17-train-170630','')
  big_df.set_value(i,'PostImage', tmp)

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:4: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
  after removing the cwd from sys.path.


In [0]:
big_df

PostImage  person  bicycle  car  motorcycle  \
0      /media/photo_828829261612072960.jpg      14        0    0           0   
1      /media/photo_828430513929064448.jpg       1        0    0           0   
2      /media/photo_828766205884276736.jpg      24        0    0           0   
3      /media/photo_828307201727029248.jpg       1        0    0           0   
4      /media/photo_828513008204804097.jpg       3        0    0           0   
5      /media/photo_828829261612072960.jpg      14        0    0           0   
6      /media/photo_828430513929064448.jpg       1        0    0           0   
7      /media/photo_828766205884276736.jpg      24        0    0           0   
8      /media/photo_828307201727029248.jpg       1        0    0           0   
9      /media/photo_828513008204804097.jpg       3        0    0           0   
10     /media/photo_828409834546593792.jpg      16        0    0           0   
11     /media/photo_828305883977351170.jpg       1        0    0           0   
12     /media/photo_828430075037151233.jpg       1        0    0           0   
13     /media/photo_828707373879422976.jpg       0        0    0           0   
14     /media/photo_828814559033032704.jpg       3        0    0           0   
15     /media/photo_827193187857006592.jpg       0        0    0           0   
16     /media/photo_827931239990587396.jpg       1        0    0           0   
17     /media/photo_828041956693471232.jpg       3        0    0           0   
18     /media/photo_827916155792871424.jpg       1        0    0           0   
19     /media/photo_827929517452828673.jpg       0        0    0           0   
20     /media/photo_827166395922317312.jpg       0        0    0           0   
21     /media/photo_827664467353808896.jpg       6        0    0           0   
22     /media/photo_827365528209596416.jpg       0        0    0           0   
23     /media/photo_827494315358302210.jpg       2        0    0           0   
24     /media/photo_827819236768698368.jpg       1        0    0           0   
25     /media/photo_827353186726006785.jpg       1        0    0           0   
26     /media/photo_827949114461712384.jpg       3        0    0           0   
27     /media/photo_827595826411274240.jpg       7        0    0           0   
28     /media/photo_827741944822501376.jpg       0        0    0           0   
29     /media/photo_826584853776887810.jpg       1        0    0           0   
...                                    ...     ...      ...  ...         ...   
11016  /media/photo_832307120376799232.jpg       1        0    0           0   
11017  /media/photo_832106054167121920.jpg      26        0    0           0   
11018  /media/photo_832048857915736065.jpg       0        0    1           0   
11019  /media/photo_832344372670652416.jpg       1        0    0           1   
11020  /media/photo_831878502051934208.jpg       2        0    1           0   
11021  /media/photo_831818855412531200.jpg       1        0    0           0   
11022  /media/photo_831573994184785920.jpg       1        0    0           0   
11023  /media/photo_831509038407970821.jpg       5        0    0           0   
11024  /media/photo_831650844848238592.jpg       2        0    0           0   
11025  /media/photo_831524904671539200.jpg       1        0    0           0   
11026  /media/photo_831746749622542336.jpg       4        0    0           0   
11027  /media/photo_831750380543410176.jpg       1        0    0           0   
11028  /media/photo_831608665320914944.jpg       5        0    0           0   
11029  /media/photo_831629121201508354.jpg       1        0    0           0   
11030  /media/photo_831351026200494080.jpg      13        0    0           0   
11031  /media/photo_831327145125257217.jpg       2        0    0           0   
11032  /media/photo_830941327789584386.jpg       1        0    0           0   
11033  /media/photo_831259464258838528.jpg       5        0    0           0   
11034  /media/photo_8312917146817

In [0]:
df2 = big_df.drop_duplicates()

In [0]:
df2.shape

(11041, 91)

In [0]:
df1.to_csv(r'/content/Drive/My Drive/Spring 2019/ML/sample for project/small_df.csv', index= False)

In [0]:
df2.to_csv(r'/content/Drive/My Drive/Spring 2019/ML/sample for project/big_df.csv', index= False)